compiler/native/compile.go

```diff
@@ -45,6 +45,12 @@ func (c *client) Compile(v interface{}) (*pipeline.Build, error) {
 		return nil, err
 	}
 
+	// inject the environment variables into the services
+	p.Secrets, err = c.EnvironmentSecrets(p.Secrets)
```

Comment:
Injects the default environment, converts parameters to env, and ensures anything in the environment block makes it into container environment for each secret container.